### PR TITLE
remove some deprecated api

### PR DESCRIPTION
**Pull request by minggo**

### CODE REVIEW

File: cocos/particle/animator/gradient-range.ts [hunk 1]
```diff
@@ -190,7 +188,7 @@ export function packGradientRange (samples: number, gr: GradientRange) {
     }
 
     const texture = new Texture2D();
-    texture.create(samples, height, PixelFormat.RGBA8888);
+    texture.reset({ width: samples, height, format: PixelFormat.RGBA8888 });
```
Comment (minggo) [inline cocos/particle/animator/gradient-range.ts:191]: will remove duck type usage in later commit

File: cocos/2d/components/deprecated.ts [hunk 2]
```diff
@@ -23,81 +23,4 @@
...
- */
-export { UIStaticBatch as UIStaticBatchComponent };
-js.setClassAlias(UIStaticBatch, 'cc.UIStaticBatchComponent');
-/**
- * Alias of [[UIOpacity]]
- * @deprecated Since v1.2
- */
-export { UIOpacity as UIOpacityComponent };
-js.setClassAlias(UIOpacity, 'cc.UIOpacityComponent');
+export {};
```
Comment (pandamicro) [inline cocos/2d/components/deprecated.ts]: I think it's better to remove this file directly

File: cocos/3d/skeletal-animation/deprecated.ts [hunk 3]
```diff
@@ -23,17 +23,4 @@
...
-import { legacyCC } from '../../core/global-exports';
-import { SkeletalAnimation } from './skeletal-animation';
-/**
- * Alias of [[SkeletalAnimation]]
- * @deprecated Since v1.2
- */
-export { SkeletalAnimation as SkeletalAnimationComponent };
-legacyCC.SkeletalAnimationComponent = SkeletalAnimation;
-js.setClassAlias(SkeletalAnimation, 'cc.SkeletalAnimationComponent');
+export {};
```
Comment (pandamicro) [inline cocos/3d/skeletal-animation/deprecated.ts]: Please remove this file directly

File: cocos/3d/skinned-mesh-renderer/deprecated.ts [hunk 4]
```diff
@@ -23,33 +23,4 @@
...
-legacyCC.SkinningModelUnit = SkinnedMeshUnit;
-js.setClassAlias(SkinnedMeshUnit, 'cc.SkinningModelUnit');
-/**
- * Alias of [[SkinnedMeshBatchRenderer]]
- * @deprecated Since v1.2
- */
-export { SkinnedMeshBatchRenderer as BatchedSkinningModelComponent };
-legacyCC.BatchedSkinningModelComponent = SkinnedMeshBatchRenderer;
-js.setClassAlias(SkinnedMeshBatchRenderer, 'cc.BatchedSkinningModelComponent');
+export {};
```
Comment (pandamicro) [inline cocos/3d/skinned-mesh-renderer/deprecated.ts]: Same

File: cocos/ui/deprecated.ts [hunk 5]
```diff
@@ -22,143 +22,4 @@
...
-export { UICoordinateTracker as UICoordinateTrackerComponent };
-js.setClassAlias(UICoordinateTracker, 'cc.UICoordinateTrackerComponent');
-/**
- * Alias of [[BlockInputEvents]]
- * @deprecated Since v1.2
- */
-export { BlockInputEvents as BlockInputEventsComponent };
-legacyCC.BlockInputEventsComponent = BlockInputEvents;
-js.setClassAlias(BlockInputEvents, 'cc.BlockInputEventsComponent');
+export {}
```
Comment (pandamicro) [inline cocos/ui/deprecated.ts]: Same here

File: cocos/2d/components/deprecated.ts (hunk 2 above)
Comment (minggo) [reply cocos/2d/components/deprecated.ts]: I think you may need it when deprecated other API again, so just keep it. If we can make sure we will not use it to deprecate API again. We can delete it.

Comment (minggo) [reply cocos/2d/components/deprecated.ts]: As discussion with @pandamicro, will use deprecated-vxxx to record deprecated functions of a specific version, so i will remove these empty files.

**Review by shrinktofit — APPROVED** [review]

File: cocos/2d/components/deprecated.ts (hunk 2 above)
Comment (minggo) [reply cocos/2d/components/deprecated.ts]: It is deleted.

File: cocos/3d/skinned-mesh-renderer/deprecated.ts (hunk 4 above)
Comment (minggo) [reply cocos/3d/skinned-mesh-renderer/deprecated.ts]: It is deleted.

File: cocos/ui/deprecated.ts (hunk 5 above)
Comment (minggo) [reply cocos/ui/deprecated.ts]: It is deleted.

File: cocos/3d/skeletal-animation/deprecated.ts (hunk 3 above)
Comment (minggo) [reply cocos/3d/skeletal-animation/deprecated.ts]: It is deleted.

**Review by SantyWang — APPROVED** [review]

**Pull request merged**